### PR TITLE
created WebDriverContainer that will serve as a universal access for …

### PR DESCRIPTION
**Pull request by Millionhat**
…the pages

### CODE REVIEW

File: src/main/java/pages/WebDriverContainer.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
+package pages;
+
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.chrome.ChromeDriver;
+import org.openqa.selenium.chrome.ChromeOptions;
+
+/**
+ * Class in charge of storing the webDriver for an easy access to the Page classes.
```
Comment (kliver98) [inline src/main/java/pages/WebDriverContainer.java]:
```suggestion
 * Class in charge of storing the webDriver for easy access to the Page classes.
```

File: src/main/java/pages/WebDriverContainer.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+   * @return WebDriver
+   */
+  public static WebDriver getInstance() {
+    if (driver == null) {
+      synchronized (WebDriverContainer.class) {
+        if (driver == null) {
+          new WebDriverContainer();
+        }
+      }
+    }
```
Comment (kliver98) [inline src/main/java/pages/WebDriverContainer.java]: Why don't just have one driver == null verification?

File: src/main/java/pages/WebDriverContainer.java [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    return driver;
+  }
+
+  public void quitDriver() {
```
Comment (kliver98) [inline src/main/java/pages/WebDriverContainer.java]: Defined but never called

File: src/test/java/BuyTshirtProcessTest.java [hunk 4]
```diff
@@ -25,7 +26,7 @@
   private SaucedomoCheckOutPage checkOut;
   private SaucedomoVerificationPage verification;
   private SaucedomoCompletedOrderPage completed;
-  private WebDriver driver;
+  private WebDriver wdc;
```
Comment (kliver98) [inline src/test/java/BuyTshirtProcessTest.java]: please use a descriptive name not just `wdc` 
also modify other files where you use it

File: src/test/java/BuyTshirtProcessTest.java [hunk 5]
```diff
@@ -73,6 +70,6 @@ public void buyTshirtTest() {
 
   @AfterMethod
   public void destroy() {
-    driver.quit();
+    //wdc.close();
```
Comment (kliver98) [inline src/test/java/BuyTshirtProcessTest.java]: this should be commented?
the method is necessary?
same comment for other files where you use it

File: src/main/java/pages/GooglePage.java [hunk 6]
```diff
@@ -13,18 +13,16 @@
  */
 public class GooglePage {
 
-  private WebDriver driver;
```
Comment (aperdomob) [inline src/main/java/pages/GooglePage.java:16]: you don't need remove this variable

File: src/test/java/BuyTshirtProcessTest.java [hunk 7]
```diff
@@ -37,17 +38,13 @@ public static void setupWebDriver() {
...
-    options.addArguments("--disable-dev-shm-usage");
-    options.addArguments("--headless");
-    driver = new ChromeDriver(options);
-    catalog = new SaucedomoCatalogPage(driver);
-    checkOut = new SaucedomoCheckOutPage(driver);
-    logIn = new SaucedomoLogInPage(driver);
-    shoppingCart = new SaucedomoShoppingCartPage(driver);
-    verification = new SaucedomoVerificationPage(driver);
-    completed = new SaucedomoCompletedOrderPage(driver);
+    driver = WebDriverContainer.getInstance();
```
Comment (aperdomob) [inline src/test/java/BuyTshirtProcessTest.java]: why do you need this instance here

**Review by aperdomob — CHANGES_REQUESTED** [review]

**Review by aperdomob — APPROVED** [review]

**Pull request merged**